.github/workflows/ci.yml

```diff
@@ -0,0 +1,85 @@
+name: CI
+
+on:
+  pull_request_target:
+    types: [opened, synchronize, reopened]
+
+concurrency:
+  group: ci-${{ github.sha }}
+  cancel-in-progress: true
+
+permissions: {}
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # tag=v3.3.0
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 0
+
+      - uses: ./github-actions/setup-bazel-remote-exec
+      - uses: actions/cache@v3
+        with:
+          path: |
+            .yarn/cache
+            .yarn/install-state.gz
+          key: v1-${{hashFiles('yarn.lock')}}
+      - run: yarn install --immutable
+      - run: yarn ng-dev format changed --check ${{ github.event.pull_request.base.sha }}
+      - run: yarn ng-dev commit-message validate-range ${{ github.event.pull_request.base.sha }} ${{ github.event.pull_request.head.sha }}
+      - run: yarn lint
+      - run: yarn tsc -p tsconfig.json
+
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # tag=v3.3.0
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - uses: ./github-actions/setup-bazel-remote-exec
+      - uses: actions/cache@v3
+        with:
+          path: |
+            .yarn/cache
+            .yarn/install-state.gz
+          key: v1-${{hashFiles('yarn.lock')}}
+      - run: yarn install --immutable
```

Comment:
Are we extracting these yarn installs + cache into a separate composite action (like in the components repo)?

Reply:
That is something we definitely can do, but in this initial testing/exploring I haven't done so.

Reply:
We should do that as a follow-up- since that will be the CI setup for dev-infra going forward